File: single-language-analysis/dataset-files/python/test_smell_001.py
import pytest

class TestUnitTest:
    def test_smell_001(self):
        locations = FileIO.read_location_file("location.csv")

        assert locations.peek_first().get_name() == "Perth"

        map = FileIO.read_distance_file("distances.csv", locations)

        assert locations.get_count() == 14
        assert map.get_node_count() == 11
        assert map.get_edge_count() == 62

        temp = map.get_edges().peek_first().find_weight("walk").get_length()
        assert temp == pytest.approx(13.1, rel=0.0001)

        temp1 = map.get_edges().peek_first().find_weight("car").get_pk_time()
        assert temp1 == 24

File: single-language-analysis/dataset-files/java/TestSmell001.java
// Original URL: https://github.com/emmai89/Assignment/blob/2ec0af36a6cde4db003a09dbf18ba1ffcf1b1ffa/UnitTest.java#L59-L78

public class TestSmell001 {
   @Test
   public void FileIOLocation()
   {
      locations = FileIO.readLocationFile("location.csv");

      assertEquals(locations.peekFirst().getName(), "Perth");

      map = FileIO.readDistanceFile("distances.csv", locations);

      assertEquals(locations.getCount(), 14);
      assertEquals(map.getNodeCount(), 11);
      assertEquals(map.getEdgeCount(), 62);

      double temp = map.getEdges().peekFirst().findWeight("walk").getLength();
      assertEquals(temp, 13.1, 0.0001);

      int temp1 = map.getEdges().peekFirst().findWeight("car").getPkTime();
      assertEquals(temp1, 24);

   }
}
